lib/errors/base-error.ts

```diff
@@ -8,10 +21,11 @@
  * The Base Error all Sequelize Errors inherit from.
  */
 class BaseError extends Error {
-  constructor(message) {
+  constructor(message: string) {
     super(message);
     this.name = 'SequelizeBaseError';
   }
 }
 
-module.exports = BaseError;
+export default BaseError;
+export { ErrorOptions, CommonErrorProperties };
```

Comment:
I'm unsure if we want to keep named exports at the bottom of the file, or if we should directly export when we create a class, function, variable etc (in the same line), as was done in `lib/utils/logger.ts`

Reply:
Aha! I'd vote for inlining it with the definition instead of at the bottom of the file

Reply:
Using inline syntax for named exports now. Though, I would keep default exports at the bottom of the file. Furthermore, I would keep the default exported object at the very bottom of the file, as seen in `validation-error.ts`

Reply:
As discussed internally, we will use inline named exports and keep default exports at the bottom of the file.